File: frontend/src/app/shared/photographer-admin.service.ts
import { Inject, Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { firstValueFrom } from 'rxjs';
import { API_BASE_URL } from './api.config';

export interface PricingProfilePayload {
  pricePerPhoto: number | null;
  bundlePrice: number | null;
  bundleSize: number | null;
  currencyCode: string | null;
}

export type PhotographerStatus =
  | 'ONBOARDING'
  | 'ACTIVE'
  | 'PAUSED'
  | 'SUSPENDED';

export type PayoutMethod =
  | 'UNSPECIFIED'
  | 'ACH'
  | 'PAYPAL'
  | 'WISE'
  | 'STRIPE_CONNECT'
  | 'MANUAL';

export interface PayoutPreferencesPayload {
  method: PayoutMethod;
  accountReference: string | null;
  payoutEmail: string | null;
  bankAccountLast4: string | null;
  bankRoutingNumber: string | null;
  taxId: string | null;
  metadata: string | null;
}

export interface CreatePhotographerRequest {
  slug: string;
  firstName: string;
  lastName: string;
  displayName: string;
  email: string;
  phoneNumber: string | null;
  studioName: string | null;
  website: string | null;
  defaultCurrency: string | null;
  status: PhotographerStatus;
  biography: string | null;
  commissionOverride: number | null;
  rateCard: PricingProfilePayload;
  payoutPreferences: PayoutPreferencesPayload;
  payoutThreshold: number | null;
  internalNotes: string | null;
}

export interface PhotographerSummary {
  id: string;
  slug: string;
  displayName: string;
  email: string;
  status: PhotographerStatus;
  defaultCurrency: string | null;
  updatedAt: string;
}

export interface PhotographerDetail extends CreatePhotographerRequest {
  id: string;
  createdAt: string;
  updatedAt: string;
}

export type UpdatePhotographerRequest = CreatePhotographerRequest;

@Injectable({ providedIn: 'root' })
export class PhotographerAdminService {
  constructor(
    private http: HttpClient,
    @Inject(API_BASE_URL) private readonly apiBaseUrl: string
  ) {}

  createPhotographer(payload: CreatePhotographerRequest): Promise<void> {
    const url = `${this.apiBaseUrl}/admin/photographers`;
    return firstValueFrom(this.http.post<void>(url, payload));
  }

  listPhotographers(): Promise<PhotographerSummary[]> {
    const url = `${this.apiBaseUrl}/admin/photographers`;
    return firstValueFrom(this.http.get<PhotographerSummary[]>(url));
  }

  getPhotographer(id: string): Promise<PhotographerDetail> {
    const url = `${this.apiBaseUrl}/admin/photographers/${id}`;
    return firstValueFrom(this.http.get<PhotographerDetail>(url));
  }

  updatePhotographer(id: string, payload: UpdatePhotographerRequest): Promise<PhotographerDetail> {
    const url = `${this.apiBaseUrl}/admin/photographers/${id}`;
    return firstValueFrom(this.http.put<PhotographerDetail>(url, payload));
  }
}

File: backend/src/main/java/com/racephotos/service/PhotographerAdminService.java
package com.racephotos.service;

import com.racephotos.auth.user.User;
import com.racephotos.auth.user.UserRepository;
import com.racephotos.domain.common.PricingProfile;
import com.racephotos.domain.photographer.PayoutMethod;
import com.racephotos.domain.photographer.PayoutPreferences;
import com.racephotos.domain.photographer.Photographer;
import com.racephotos.domain.photographer.PhotographerRepository;
import com.racephotos.domain.photographer.PhotographerStatus;
import com.racephotos.service.dto.CreatePhotographerCommand;
import com.racephotos.service.dto.PayoutPreferencesData;
import com.racephotos.service.dto.PricingProfileData;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import org.springframework.dao.DataIntegrityViolationException;
import org.springframework.http.HttpStatus;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;
import org.springframework.web.server.ResponseStatusException;

import java.math.BigDecimal;
import java.util.Objects;
import java.util.UUID;

@Service
public class PhotographerAdminService {

    private static final Logger log = LogManager.getLogger(PhotographerAdminService.class);

    private final PhotographerRepository photographerRepository;
    private final UserRepository userRepository;

    public PhotographerAdminService(
            PhotographerRepository photographerRepository,
            UserRepository userRepository
    ) {
        this.photographerRepository = Objects.requireNonNull(photographerRepository, "photographerRepository");
        this.userRepository = Objects.requireNonNull(userRepository, "userRepository");
    }

    @Transactional
    public Photographer createPhotographer(CreatePhotographerCommand command) {
        if (command == null) {
            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Request body is required");
        }

        UUID createdById = command.createdByUserId();
        if (createdById == null) {
            throw new ResponseStatusException(HttpStatus.UNAUTHORIZED, "User session is required");
        }

        User creator = userRepository.findById(createdById)
                .orElseThrow(() -> new ResponseStatusException(HttpStatus.UNAUTHORIZED, "User session is invalid"));

        String slug = normalize(command.slug());
        if (slug == null) {
            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Slug must not be blank");
        }
        if (photographerRepository.existsBySlug(slug)) {
            throw new ResponseStatusException(HttpStatus.CONFLICT, "A photographer with this slug already exists");
        }

        String email = normalizeEmail(command.email());
        if (email == null) {
            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Email must not be blank");
        }
        if (photographerRepository.existsByEmail(email)) {
            throw new ResponseStatusException(HttpStatus.CONFLICT, "A photographer with this email already exists");
        }

        BigDecimal commissionOverride = normalizeDecimal(command.commissionOverride());
        if (commissionOverride != null &&
                (commissionOverride.compareTo(BigDecimal.ZERO) < 0 || commissionOverride.compareTo(BigDecimal.ONE) > 0)) {
            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Commission override must be between 0 and 1");
        }

        BigDecimal payoutThreshold = normalizeDecimal(command.payoutThreshold());
        if (payoutThreshold != null && payoutThreshold.compareTo(BigDecimal.ZERO) < 0) {
            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Payout threshold must be non-negative");
        }

        Photographer photographer = new Photographer();
        photographer.setSlug(slug);
        photographer.setFirstName(normalize(command.firstName()));
        photographer.setLastName(normalize(command.lastName()));
        photographer.setDisplayName(normalize(command.displayName()));
        photographer.setEmail(email);
        photographer.setPhoneNumber(normalize(command.phoneNumber()));
        photographer.setStudioName(normalize(command.studioName()));
        photographer.setWebsite(normalize(command.website()));
        photographer.setDefaultCurrency(normalizeUpper(command.defaultCurrency()));
        photographer.setStatus(command.status() == null ? PhotographerStatus.ONBOARDING : command.status());
        photographer.setBiography(normalize(command.biography()));
        photographer.setCommissionOverride(commissionOverride);
        photographer.setPayoutThreshold(payoutThreshold);
        photographer.setInternalNotes(normalize(command.internalNotes()));
        photographer.setCreatedBy(creator);

        applyRateCard(photographer, command.rateCard());
        applyPayoutPreferences(photographer, command.payoutPreferences());

        try {
            Photographer saved = photographerRepository.save(photographer);
            log.info("User {} created photographer {} with slug '{}'", creator.getEmail(), saved.getId(), saved.getSlug());
            return saved;
        } catch (DataIntegrityViolationException e) {
            log.warn("Failed to create photographer with slug '{}' due to constraint violation", slug);
            throw new ResponseStatusException(HttpStatus.CONFLICT, "Photographer could not be created", e);
        }
    }

    private void applyRateCard(Photographer photographer, PricingProfileData data) {
        if (data == null) {
            return;
        }
        PricingProfile profile = photographer.getRateCard();
        if (profile == null) {
            profile = new PricingProfile();
            photographer.setRateCard(profile);
        }
        profile.setPricePerPhoto(data.pricePerPhoto());
        profile.setBundlePrice(data.bundlePrice());
        profile.setBundleSize(data.bundleSize());
        profile.setCurrencyCode(normalizeUpper(data.currencyCode()));
    }

    private void applyPayoutPreferences(Photographer photographer, PayoutPreferencesData data) {
        if (data == null) {
            return;
        }
        PayoutPreferences preferences = photographer.getPayoutPreferences();
        if (preferences == null) {
            preferences = new PayoutPreferences();
            photographer.setPayoutPreferences(preferences);
        }
        preferences.setMethod(data.method() == null ? PayoutMethod.UNSPECIFIED : data.method());
        preferences.setAccountReference(normalize(data.accountReference()));
        preferences.setPayoutEmail(normalizeEmail(data.payoutEmail()));
        preferences.setBankAccountLast4(normalize(data.bankAccountLast4()));
        preferences.setBankRoutingNumber(normalize(data.bankRoutingNumber()));
        preferences.setTaxId(normalize(data.taxId()));
        preferences.setMetadata(normalize(data.metadata()));
    }

    private String normalize(String value) {
        if (value == null) {
            return null;
        }
        String trimmed = value.trim();
        return trimmed.isEmpty() ? null : trimmed;
    }

    private String normalizeUpper(String value) {
        String normalized = normalize(value);
        return normalized == null ? null : normalized.toUpperCase();
    }

    private String normalizeEmail(String email) {
        String normalized = normalize(email);
        return normalized == null ? null : normalized.toLowerCase();
    }

    private BigDecimal normalizeDecimal(BigDecimal value) {
        return value;
    }
}
